### PR TITLE
add new rule changedParameterOrder

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

File: openapi-diff/src/modeler/AutoRest.Swagger/Model/Operation.cs [hunk 1]
```diff
@@ -158,9 +158,28 @@ private void CheckParameters(ComparisonContext<ServiceDefinition> context, Opera
...
+            // Check whether operation parameter order change
+            for (int i = 0; i < Parameters.Count; i++)
+            {
+                for (int j = i + 1; j < Parameters.Count; j++)
+                {
+                    var priorI = FindParameterIndex(Parameters.ElementAt(i), priorOperation.Parameters);
+                    var priorJ = FindParameterIndex(Parameters.ElementAt(j), priorOperation.Parameters);
+                    if (priorI != -1 && priorJ != -1 && priorI > priorJ)
+                    {
+                        context.LogBreakingChange(ComparisonMessages.ChangedParameterOrder, Parameters.ElementAt(i).Name, Parameters.ElementAt(j).Name);
```
Comment (jianyexi) [inline openapi-diff/src/modeler/AutoRest.Swagger/Model/Operation.cs:178]: Could you add a doc for this new rule ?

Comment (ruowan) [reply openapi-diff/src/modeler/AutoRest.Swagger/Model/Operation.cs]: Added doc.

File: openapi-diff/src/modeler/AutoRest.Swagger/ComparisonMessages.cs [hunk 2]
```diff
@@ -90,6 +90,14 @@ public static class ComparisonMessages
             Type = MessageType.Removal
         };
 
+        public static MessageTemplate ChangedParameterOrder = new MessageTemplate
+        {
+            Id = 1042,
+            Code = nameof(ComparisonMessages.ChangedParameterOrder),
+            Message = "The parameter order '{0}' was changed. ",
```
Comment (jianyexi) [inline openapi-diff/src/modeler/AutoRest.Swagger/ComparisonMessages.cs:97]: Can we use  "The order of parameter '{0}' was changed."

File: docs/rules/1042.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+### 1042 - ChangedParameterOrder
+
+**Description**: Check whether operation parameter order is changed. If operation parameter order changed, it cause SDK breaking change.
```
Comment (jianyexi) [inline docs/rules/1042.md:3]: "it causes"

**Review by jianyexi — APPROVED** [review]

**Pull request merged**